common/analysis/file_analyzer_test.cc

```diff
@@ -63,13 +63,34 @@ TEST(FileAnalyzerTest, TokenErrorMessageSameLine) {
     EXPECT_EQ(message, "token: \"w0rld\" at 2:5-9");
   }
   {
+    constexpr bool with_diagnostic_context = false;
     const auto message = analyzer.LinterTokenErrorMessage(
-        {error_token, AnalysisPhase::kParsePhase});
+        {error_token, AnalysisPhase::kParsePhase}, with_diagnostic_context);
     EXPECT_TRUE(absl::StrContains(
         message, "hello.txt:2:5: syntax error, rejected \"w0rld\""));
   }
 }
 
+// Verify that an error token on one line is reported correctly.
+TEST(FileAnalyzerTest, TokenErrorMessageSameLineWithContext) {
+  const std::string text("hello, world\nbye w0rld\n");
+  FakeFileAnalyzer analyzer(text, "hello.txt");
+  const TokenInfo error_token(1, analyzer.Data().Contents().substr(17, 5));
+  {
+    const auto message = analyzer.TokenErrorMessage(error_token);
+    EXPECT_EQ(message, "token: \"w0rld\" at 2:5-9");
+  }
+  {
+    constexpr bool with_diagnostic_context = true;
+    const auto message = analyzer.LinterTokenErrorMessage(
+        {error_token, AnalysisPhase::kParsePhase}, with_diagnostic_context);
+    EXPECT_TRUE(
+        absl::StrContains(message,
+                          "hello.txt:2:5: syntax error, rejected \"w0rld\" "
```

Comment:
In general, in this test we want to show that whenever we have an error, we show the context and an arrow that hightlights where things are problematic.
So our tests should try to capture that.

I'd probably break this up so that the newlines align, so that it is easy to see that error message and 'arrow', that should point to "w0rld" above, line up.
If you break after the parenthesis, you should get more space to do so:

```
EXPECT_TRUE(absl::StrContains(
   message,
   ".......\n"
   " ...^...");
```

Reply:
Nice, now it is easy to see what is marked.